toolkit/tools/imagegen/configuration/packageRepo.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Parser for the image builder's PackageRepo configuration schemas.
+
+package configuration
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"microsoft.com/pkggen/internal/file"
+	"microsoft.com/pkggen/internal/logger"
+	"microsoft.com/pkggen/internal/network"
+	"microsoft.com/pkggen/internal/safechroot"
+	"microsoft.com/pkggen/internal/shell"
+)
+
+// PackageRepo defines the RPM repo to pull packages from during the installation
+// or after the system is installed. The "Install" option indicates that the provided
+// repository configuration will be saved in the installed system if specified, and only
+// available during the installation process if not
+type PackageRepo struct {
+	Name    string `json:"Name"`
+	BaseUrl string `json:"BaseUrl"`
+	Install bool   `json:"Install"`
```

Comment:
Related to the non-iso builds, do we want the user to be able to add a repo here without making it available during install? IE, `InstallIntoImage bool` and `UseDuringInstall bool`. (If so, maybe add a test that complains in both are false?).

Reply:
Not enabling this feature for non-ISO builds for now